pyrometheus/codegen/python.py

```diff
@@ -321,6 +327,33 @@ def get_mixture_internal_energy_mass(self, temperature, mass_fractions):
         emix = self.get_mass_average_property(mass_fractions, e0_rt)
         return self.gas_constant * temperature * emix
 
+    def get_mixture_entropy_mass(self, pressure, temperature, mass_fractions):
+        mmw = self.get_mix_molecular_weight(mass_fractions)
+        return 1.0/mmw * self.get_mixture_entropy_mole(pressure, temperature, <%
+                                                       %>mass_fractions)
+
+    def get_mole_average_property(self, mass_fractions, spec_property):
+        mmw = self.get_mix_molecular_weight(mass_fractions)
+        mole_fracs = self.get_mole_fractions(mmw, mass_fractions)
+        return sum([mole_fracs[i] * spec_property[i]
```

Comment:
Should this not be a `usr_np.sum`?

Reply:
In this case maybe. I say that because some np.sum were yielding wrong results in mirge-com: it would sum over all elements of all arrays and return a single number instead of an array. *This is something we have to check in pyrometheus with grids!*

Reply:
`self.usr_np.sum(..., axis=0)`?